Library/Entitlement.cs

```diff
@@ -0,0 +1,130 @@
+        }
+
+        public Entitlement(Account account) : this()
+        {
+            AccountCode = account.AccountCode;
+        }
+
+        internal Entitlement(XmlTextReader xmlReader)
+        {
+            ReadXml(xmlReader);
+        }
+
+        public Entitlement() { }
+
+        /// <summary>
+        /// Lookup a Recurly account's entitlements
+        /// </summary>
+        /// <param name="accountCode"></param>
+        /// <returns></returns>
+        //public static Entitlement Get(string accountCode)
+        //{
+        //    if (string.IsNullOrWhiteSpace(accountCode))
+        //    {
+        //        return null;
+        //    }
+
+        //    var entitlement = new Entitlement();
+
+        //    var statusCode = Client.Instance.PerformRequest(Client.HttpRequestMethod.Get,
+        //        EntitlementUrl(accountCode),
+        //        entitlement.ReadXml);
+
+        //    return statusCode == HttpStatusCode.NotFound ? null : entitlement;
+        //}
+
+        //private static string EntitlementUrl(string accountCode)
+        //{
+        //    return UrlPrefix + Uri.EscapeDataString(accountCode) + UrlPostfix;
+        //}
+
+        internal override void ReadXml(XmlTextReader reader)
+        {
+            while (reader.Read())
+            {
+                if (reader.Name == "entitlement" && reader.NodeType == XmlNodeType.EndElement)
+                    break;
+
+                if (reader.NodeType != XmlNodeType.Element) continue;
+
+                switch (reader.Name)
+                {
+                    case "entitlement":
+                        // The element's opening tag - nothing to do
+                        break;
+
+                    case "customer_permission":
+                        CustomerPermission = new CustomerPermission(reader);
+                        break;
+
+                    case "granted_by":
+                        ReadXMLGrantedBys(reader);
+                        //GrantedBy = new GrantedBy(reader);
+                        break;
+
+                    case "created_at":
+                        CreatedAt = reader.ReadElementContentAsDateTime();
+                        break;
+
+                    case "updated_at":
+                        UpdatedAt = reader.ReadElementContentAsDateTime();
+                        break;
+                }
+            }
+        }
+
+
+        internal void ReadXMLGrantedBys(XmlTextReader reader)
+        {
+            while (reader.Read())
+            {
+                if (reader.Name == "granted_by" && reader.NodeType == XmlNodeType.EndElement)
+                    break;
+
+                if (reader.NodeType != XmlNodeType.Element) continue;
+
+
+                if (reader.NodeType == XmlNodeType.Element)
+                {
+                    var href = reader.GetAttribute("href");
+                    string id = href.Substring(href.LastIndexOf("/") + 1);
+                    GrantedBy.Add(reader.Name, id);
```

Comment:
@amandamfielding a Dictionary in c# need a unique keys, so if you had multiple subscriptions and "subscription" is the key like it is here, it would break.  We could solve that by making the `id` the key instead of `reader.Name` (just swap `reader.Name` and `id` in the above line), OR instead of using a dictionary we could use a generic `KeyValuePair<string, string>` type.